app/jobs/scheduled/redeliver_web_hook_events.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+require "excon"
+
+module Jobs
+  class RedeliverWebHookEvents < ::Jobs::Scheduled
+    every 1.minute
+
+    sidekiq_options queue: "low"
+    sidekiq_options retry: false
+
+    REDELIVERED = "redelivered"
+    LIMIT = 20
+
+    def execute(args)
+      redelivery_events = RedeliveringWebhookEvent.includes(web_hook_event: :web_hook).limit(LIMIT)
+      redelivery_events.each do |redelivery_event|
```

Comment:
The main concern here now would be an instance of this job taking > 1 minute to complete, which is definitely possible. Then you have the job run again in the next minute, and the first job may have sent 10 + deleted 10 redelivery events, the second job will start from 10 to get the next 20, so you will have overlap.

Probably the easiest solution here is to add a `status` to `RedeliveringWebhookEvent` (default `pending`). Then, you do:

```ruby
redelivery_events = RedeliveringWebhookEvent.where(status: :pending).includes(web_hook_event: :web_hook).limit(LIMIT)
redelivery_events.update_all(status: :processing)
```

At the start of the job.  At the end of the job, any of these that didn't get deleted you could set `status: :pending` to put them back in the pool (or if you go with my other comment, you wouldn't need to since they are all deleted).

Reply:
@rngus2344 you would still need to do this `update_all/where` even though you changed to delete on failure, otherwise the next job could potentially pick up ones you could operate on.